self-hosting/README.md

```diff
@@ -0,0 +1,310 @@
+To find out which service is using a conflicting port: On Linux:
+
+`sudo netstat -plnt | grep ':<port>'`
+
+On Mac (10.10 and above):
+
+`sudo lsof -iTCP -sTCP:LISTEN -n -P | grep ':<port>'` 
+
+You can either kill the service which is occupying HTTP/HTTPS ports, or run the container with forwarded ports that are free. In your compose file, change the ports under medic-os:
+
+```yaml
+
+services:
+  medic-os:
+    container_name: medic-os
+    image: medicmobile/medic-os:cht-3.7.0-rc.1
+    volumes:
+      - medic-data:/srv
+    ports:
+     - 8080:80
+     - 444:443
+
+```
+
+Turn off and remove all existing containers that were started:
+
+ `sudo docker-compose -f /path/to/docker-compose.yml down`
+
+Bring Up the containers in detached mode with the new forwarded ports.
+
+ `sudo docker-compose -f /path/to/docker-compose.yml up -d`
+
+Note: You can  substitute 8080, 444 with whichever ports are free on your host. You would now visit https://localhost:444 to visit your project.
+
+### Helpful Docker Commands
+
+#### Get  into a container and view specific service logs 
+
+`docker exec -it medic-os /bin/bash`
+
+##### Once inside container
+
+- view couchdb logs:
+`less /srv/storage/medic-core/couchdb/logs/startup.log`
+- view medic-api logs:
+`less /srv/storage/medic-api/logs/medic-api.log`
+- view medic-sentinel logs:
+`less /srv/storage/medic-sentinel/logs/medic-sentinel.log`
+
+#### View container stderr/stdout log
+
+`docker logs medic-os`
+`docker logs haproxy`
+
+#### Other handy commands
+
+- list running containers `docker ps`
+
+- list all available docker containers with their status `docker ps -a`
+
+- stop container `docker stop <container_id>`
+
+- start container `docker start <container_id>`
+
+- list all stoped containers `docker ps -f "status=exited"`
+
+## SSL Certificate Installation
+
+Once your application is up,  you can install your ssl certificates by following the steps below.
+
+- First you need to have your ssl certificate files handy. Both the key file (`ssl.key`) that you used to generate the certificate and the certificate file (`ssl.crt`) issued by certification authority e.g. letsencrypt. 
+
+Note: If your certification authority issued you `.pem` files, you can easily convert `.pem` files to `.crt` using the command below. 
+
+```bash
+openssl x509 -outform der -in your-cert.pem -out your-cert.crt
+
+```
+
+### Copy certificates into the medic-os container
+
+```bash
+sudo docker cp /path/to/ssl.crt medic-os:/srv/settings/medic-core/nginx/private/ssl.crt
+sudo docker cp /path/to/ssl.key medic-os:/srv/settings/medic-core/nginx/private/ssl.key
+```
+
+### Edit the nginx configuration file and restart
+
+```bash
+sudo docker exec -it medic-os /bin/bash
+sed -i "s|default.crt|ssl.crt|" /srv/settings/medic-core/nginx/nginx.conf
+sed -i "s|default.key|ssl.key|" /srv/settings/medic-core/nginx/nginx.conf
+
+# enter medic-os container:
+sudo docker exec -it medic-os /bin/bash
+ /boot/svc-restart medic-core nginx
+
+```
```

Comment:
Last LAST change ;)  

I think it's worth collapsing these as  you don't need to run `docker exec -it medic-os /bin/bash` twice, just once:

```suggestion
```bash
# enter medic-os container, update configs and restart nginx
sudo docker exec -it medic-os /bin/bash
sed -i "s|default.crt|ssl.crt|" /srv/settings/medic-core/nginx/nginx.conf
sed -i "s|default.key|ssl.key|" /srv/settings/medic-core/nginx/nginx.conf
/boot/svc-restart medic-core nginx
```
```

Reply:
Addressed